include/zephyr/bluetooth/audio/cap.h

```diff
@@ -46,6 +48,371 @@ extern "C" {
+	 *                  Coordinated Set Identification Service.
+	 */
+	void (*discovery_complete)(struct bt_conn *conn, int err,
+				   const struct bt_csis_client_csis_inst *csis_inst);
+
+	/**
+	 * @brief Callback for bt_cap_initiator_unicast_audio_start().
+	 *
+	 * @param unicast_group  The unicast group pointer supplied to
+	 *                       bt_cap_initiator_unicast_audio_start().
+	 * @param err            0 if success, else BT_GATT_ERR() with a
+	 *                       specific ATT (BT_ATT_ERR_*) error code.
+	 * @param conn           Pointer to the connection where the error
+	 *                       occurred. NULL if @p err is 0.
+	 */
+	void (*unicast_start_complete)(struct bt_audio_unicast_group *unicast_group,
+				       int err, struct bt_conn *conn);
+
+	/**
+	 * @brief Callback for bt_cap_initiator_unicast_audio_update().
+	 *
+	 * @param unicast_group  The unicast group pointer supplied to
+	 *                       bt_cap_initiator_unicast_audio_update().
+	 * @param err            0 if success, else BT_GATT_ERR() with a
+	 *                       specific ATT (BT_ATT_ERR_*) error code.
+	 * @param conn           Pointer to the connection where the error
+	 *                       occurred. NULL if @p err is 0.
+	 */
+	void (*unicast_update_complete)(struct bt_audio_unicast_group *unicast_group,
+					int err, struct bt_conn *conn);
+
+	/**
+	 * @brief Callback for bt_cap_initiator_unicast_audio_stop().
+	 *
+	 * @param unicast_group  The unicast group pointer supplied to
+	 *                       bt_cap_initiator_unicast_audio_stop().
+	 * @param err            0 if success, else BT_GATT_ERR() with a
+	 *                       specific ATT (BT_ATT_ERR_*) error code.
+	 * @param conn           Pointer to the connection where the error
+	 *                       occurred. NULL if @p err is 0.
+	 */
+	void (*unicast_stop_complete)(struct bt_audio_unicast_group *unicast_group,
+				      int err, struct bt_conn *conn);
+#endif /* CONFIG_BT_AUDIO_UNICAST_CLIENT */
+};
+
+/**
+ * @brief Discovers audio support on a remote device.
+ *
+ * This will discover the Common Audio Service (CAS) on the remote device, to
+ * verify if the remote device supports the Common Audio Profile.
+ *
+ * @param conn Connection to a remote server.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int bt_cap_initiator_unicast_discover(struct bt_conn *conn);
+
+/** Type of CAP set */
+enum bt_cap_set_type {
+	/** The set is an ad-hoc set */
+	BT_CAP_SET_TYPE_AD_HOC,
+	/** The set is a CSIP Coordinated Set */
+	BT_CAP_SET_TYPE_CSIP,
+};
+
+/** Represents a Common Audio Set member that are either in a Coordinated or ad-hoc set */
+struct bt_cap_set_member {
+	union {
+		/** @brief Connection pointer if the type is BT_CAP_SET_TYPE_AD_HOC. */
+		struct bt_conn *member;
+
+		/** CSIP Coordinated Set struct used if type is BT_CAP_SET_TYPE_CSIP. */
+		struct bt_csis_client_set_member *csip;
+	};
+};
```

Comment:
Looks like this is just a union instead
```suggestion
union bt_cap_set_member {
	/** @brief Connection pointer if the type is BT_CAP_SET_TYPE_AD_HOC. */
	struct bt_conn *member;

	/** CSIP Coordinated Set struct used if type is BT_CAP_SET_TYPE_CSIP. */
	struct bt_csis_client_set_member *csip;
};
```

Reply:
Good point :) 

This struct might even get removed as it is only used for the start unicast parameters and the `bt_cap_broadcast_to_unicast_param`, but I think the latter may switch to use the unicast group instead.